crates/jiter-python/tests/test_jiter.py

```diff
@@ -309,10 +334,36 @@ def test_unicode_roundtrip_ensure_ascii():
 
 
 def test_catch_duplicate_keys():
-    assert jiter.from_json(b'{"foo": 1, "foo": 2}') == {"foo": 2}
+    assert jiter.from_json(b'{"foo": 1, "foo": 2}') == {'foo': 2}
 
-    with pytest.raises(ValueError, match='Detected duplicate key "foo" at line 1 column 18'):
+    with pytest.raises(
+        ValueError, match='Detected duplicate key "foo" at line 1 column 18'
+    ):
         jiter.from_json(b'{"foo": 1, "foo": 2}', catch_duplicate_keys=True)
 
-    with pytest.raises(ValueError, match='Detected duplicate key "foo" at line 1 column 28'):
+    with pytest.raises(
+        ValueError, match='Detected duplicate key "foo" at line 1 column 28'
+    ):
         jiter.from_json(b'{"foo": 1, "bar": 2, "foo": 2}', catch_duplicate_keys=True)
+
+
+def test_against_json():
+    for data in JITER_BENCH_DATAS:
+        assert jiter.from_json(data) == json.loads(data)
+
+
+def test_multithreaded_parsing():
+    """Basic sanity check that running a parse in multiple threads is fine."""
+    expected_datas = [json.loads(data) for data in JITER_BENCH_DATAS]
+
+    def assert_jiter_ok(data: bytes, expected: Any) -> bool:
+        return jiter.from_json(data) == expected
+
+    with ThreadPoolExecutor(8) as pool:
+        results = []
+        for _ in range(1000):
+            for data, expected_result in zip(JITER_BENCH_DATAS, expected_datas):
+                results.append(pool.submit(assert_jiter_ok, data, expected_result))
+
+        for result in results:
+            assert result.result()
```

Comment:
Can confirm with this simple test that the `Mutex` basically stops parallelism when the cache is enabled, using `jiter.from_json(data, cache_mode="none")` leads to about 8x speedup on my machine.

I still don't mind that really, this PR just gets the freethreaded mode working and we can worry about that sort of optimization later.